sqlmesh/core/engine_adapter/base.py

```diff
@@ -407,6 +409,44 @@ def create_index(
         )
         self.execute(expression)
 
+    def _pop_creatable_type_from_properties(
+        self,
+        properties: t.Dict[str, exp.Expression],
+    ) -> t.Optional[exp.Property]:
+        """Pop out the creatable_type from the properties dictionary (if exists (return it/remove it) else return none).
+        It also checks that none of the expressions are MATERIALIZE as that conflicts with the `materialize` parameter.
+        """
+        for key in list(properties.keys()):
+            upper_key = key.upper()
+            if upper_key == KEY_FOR_CREATABLE_TYPE:
+                value = properties.pop(key)
+                if (
+                    isinstance(value, exp.Column)
+                    and isinstance(value.this, exp.Identifier)
+                    and isinstance(value.this.this, str)
+                ):
```

Comment:
I don't think we're ever _not_ going to have a `Column` here, hence why I went directly for `...pop(key).name` in that other [comment](https://github.com/TobikoData/sqlmesh/pull/3253#discussion_r1811270890).